src/lib/work_partitioner/pending_zkapp_command.ml

```diff
@@ -42,13 +42,13 @@ let generate_merge ~(t : t) () =
   let%map proof1, proof2 = try_take2 t.pending_mergeable_proofs in
   Spec.Sub_zkapp.Merge { proof1; proof2 }
 
-let generate_segment ~(t : t) () =
+let next_segment (t : t) =
```

Comment:
Let's also add a short doc explaining that it dequeues from `unscheduled_segments`.
I want it to be clear that the function changes the state.